disco/extensions/upgrade_simulation/upgrades/common_functions.py

```diff
@@ -42,8 +44,36 @@
                     0: "none"  # 0 maps to none, which means impedance units and line length units match
                     }
 
+
+class CircuitSolveParams(BaseModel):
+    raise_exception: bool = True
+    calcvoltagebases: bool = False
+    enable_pydss_controllers: bool = False
+    pydss_controller_manager: Any = None
+
+
+class ReloadCircuitParams(BaseModel):
+    dc_ac_ratio: bool = None
+    max_control_iterations: int = None
+    pydss_volt_var_model: Any = None
+
+
+class SimulationParams(BaseModel):
+    timepoint_multipliers: Dict = None

```

Comment:
Would it be better for this to default to an empty dict?

Reply:
If defined, then it is a dict, else it's None (and not used)
So is it fine to keep it None?

Reply:
In general I think it's easier to reason about these objects if the type is fixed. Defaulting to an empty dict allows you to do that. You can always check if it's empty as a boolean, which is probably similar to what you're doing:
```
if not model.timepoint_multipliers:
    # not set
```

Reply:
Okay makes sense, I initiated it as an empty dict in both places.